lib/graphql/input_object_type.rb

```diff
@@ -27,6 +27,32 @@ def valid_non_null_input?(input)
       input_fields.all? { |name, field| field.type.valid_input?(input[name]) }
   end
 
+  def validate_non_null_input(input)
+    # Inputs are only required to provide an `all?` method and a `[]` method. That's not enough
```

Comment:
It's ok to extend the requirements here -- the real reason I added that was to make sure there wasn't any hard type-checking. 

Previously, it required that inputs were a `Hash`, but that didn't play nice when inputs came as `ActionController::Params`, which I definitely want to support! 

So, if you want to add a few common methods to that "minimum requirements", please do!